profiler/src/main/java/com/splunk/opentelemetry/profiler/JFR.java

```diff
@@ -61,4 +70,40 @@ public RecordedEvent readEvent(RecordingFile file, Path path) {
       throw new JfrException("Error reading events from " + path, e);
     }
   }
+
+  public void setStackDepth(int stackDepth) {
+    try {
+      JfrAccess.init();
+      int currentDepth = Options.getStackDepth();
+      // we only increase stack depth
+      if (currentDepth < stackDepth) {
```

Comment:
If user has set a large stack depth from command line we won't change it to a smaller one. I don't know whether this is necessary.

Reply:
This thread can be resolved